x-pack/plugins/secops/server/lib/events/elasticsearch_adapter.ts

```diff
@@ -31,37 +32,47 @@ export class ElasticsearchEventsAdapter implements EventsAdapter {
             count: item.doc_count,
           }))
         : [];
+    const { limit } = options.pagination;
+    const totalCount = getOr(0, 'hits.total.value', response);
     const hits = response.hits.hits;
-    const events = hits.map(hit => formatEventsData(options.fields, hit, eventFieldsMap)) as [
-      EventItem
-    ];
+    const eventsEdges: EcsEdges[] = hits.map(hit =>
+      formatEventsData(options.fields, hit, eventFieldsMap)
+    );
+    const hasNextPage = eventsEdges.length === limit + 1;
+    const edges = hasNextPage ? eventsEdges.splice(0, limit) : eventsEdges;
+    const lastCursor = get('cursor', edges.slice(-1)[0]);
```

Comment:
optional, I think you can use lodash `last` like so: `const lastCursor = get('cursor', last(edges));` to remove some of the syntax noise